### PR TITLE
[WIP] Add a test 'behavior' pattern for testing different Lock configs

### PR DESCRIPTION
**Pull request by NickCuso**
# Description

Adding a behavior pattern for testing various lock configs, and port one of our existing tests over.  Let me know what you think about the approach.  If good, we can port some more tests over time (ideally solidity-coverage will help inform where the gaps are) and we can aim to have new tests use behaviors

# Issues

Fixes #
Refs https://github.com/unlock-protocol/unlock/issues/2174

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [x] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [x] My code follows the style guidelines of this project, including naming conventions
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [x] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: smart-contracts/test/Lock/behaviors/getBalance.behavior.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+function getBalanceBehavior(options) {
+  describe('Lock / behaviors / getBalance', () => {
+    let accounts, lock, keyPrice
+
+    beforeEach(async () => {
+      ;({ accounts, lock } = options)
```
Comment (nfurfaro) [inline smart-contracts/test/Lock/behaviors/getBalance.behavior.js]: What is the leading `;` for?

**Comment by nfurfaro** [comment]
I _think_ this approach looks good. Can you outline the rationale behind it for context? I t would help me to assess more clearly.

File: smart-contracts/test/Lock/behaviors/getBalance.behavior.js (hunk 1 above)
Comment (NickCuso) [reply smart-contracts/test/Lock/behaviors/getBalance.behavior.js]: It's from eslint... but yea it creates a strange syntax. I was trying a few different approaches while putting this together, I'll go back to see if we can change this one a bit.

**Comment by NickCuso** [comment]
I'm going to try reworking this a little, and address the Unlock side (from #5690)

**Comment by NickCuso** [comment]
k finally updated

**Pull request merged**